crypto/secp256r1/verifier.go

```diff
@@ -0,0 +1,39 @@
+package secp256r1
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"errors"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var (
+	secp256k1halfN = new(big.Int).Div(elliptic.P256().Params().N, big.NewInt(2))
+)
+
+// Verifies the given signature (r, s) for the given hash and public key (x, y).
+func Verify(hash []byte, r, s, x, y *big.Int) ([]byte, error) {
```

Comment:
Thanks again, I did not realize that returning an error will revert all of the transaction.

I think I can proceed with returning `true` for the valid signatures and `false` for the invalid signatures and other errors, also I don't think it would be any problem not to return these errors from the `Verify` function.

My question is for the alternative method: is returning invalid signatures as an `error` logically wrong (returning only `nil` as an `error` for the valid signatures)?

Reply:
I think that, for the caller of `Verify`, it should not really matter whether the signature is 'invalid' in one way or the other way.

Reply:
Specifically, we have three failure conditions here:

- The public key is invalid.
- The signature does not match the key.
- Signature is malleable.

I think that it doesn't matter for the caller which of these conditions happened. If any of these conditions is true, the signature is invalid.

Reply:
Thanks, resolving the conversation!